### PR TITLE
[Mobile Payments] Remove legacy collect payment

### PR DESCRIPTION
**Pull request by joshheald**
Part of: #9390 

## Description

In sTAP Away, we took the opportunity to duplicate and refactor the In Person Payments code.

Now that the new code is in use, this PR removes the legacy code, and ports the tests (where possible) to target the new code.

Test coverage could still be improved further.

## Testing instructions

**Consider not testing this PR individually** – tests on #10531 should suffice

Take test payments using Tap to Pay and an external card reader, using both the Order Creation and Simple Payments flows.

Tests should pass

## Screenshots

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -105,8 +105,8 @@
         "repositoryURL": "https://github.com/woocommerce/woocommerce-shared.git",
         "state": {
           "branch": null,
-          "revision": "b68d4c71a2a2b7ba1af2e683f04dffe563fc6194",
-          "version": "0.6.0"
+          "revision": "03d9c94279361e0b002055b9fff3f75ca51ed33e",
+          "version": "0.5.0"
```
Comment (iamgabrielma) [inline WooCommerce.xcworkspace/xcshareddata/swiftpm/Package.resolved]: This looks like needs to be reverted, or merging trunk should resolve it:
```suggestion
          "revision": "b68d4c71a2a2b7ba1af2e683f04dffe563fc6194",
          "version": "0.6.0"
```

File: WooCommerce/WooCommerceTests/Mocks/MockCardPresentPaymentPreflightController.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import Foundation
+@testable import WooCommerce
+import Yosemite
+import Combine
+import Fakes
```
Comment (iamgabrielma) [inline WooCommerce/WooCommerceTests/Mocks/MockCardPresentPaymentPreflightController.swift]: I believe these are already linked through Yosemite, so we can remove the import:
```suggestion
```

File: WooCommerce/WooCommerceTests/Mocks/MockCardPresentPaymentPreflightController.swift [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+final class MockCardPresentPaymentPreflightController: CardPresentPaymentPreflightControllerProtocol {
+    func start(discoveryMethod: CardReaderDiscoveryMethod?) async {
+
```
Comment (iamgabrielma) [inline WooCommerce/WooCommerceTests/Mocks/MockCardPresentPaymentPreflightController.swift:8]: I know we're not executing any logic here in the mock's `start()`, but feels odd to have an async method that does not do anything. I have no suggestions really (perhaps a mock scenario that confirms discovery/connected readers?) but I thought was worth to bring it up.

File: WooCommerce/WooCommerceTests/Mocks/MockCollectOrderPaymentAnalyticsTracker.swift [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import Foundation
+@testable import WooCommerce
+import Yosemite
+
+final class MockCollectOrderPaymentAnalyticsTracker: CollectOrderPaymentAnalyticsTracking {
+    var connectedReaderModel: String?
+
+    func preflightResultRecieved(_ result: CardReaderPreflightResult?) {
```
Comment (iamgabrielma) [inline WooCommerce/WooCommerceTests/Mocks/MockCollectOrderPaymentAnalyticsTracker.swift]: As a non-native feel free to ignore this, should it be `preflightResultReceived` ?

**Comment by iamgabrielma** [comment]
I left it un-approved because of [this](https://github.com/woocommerce/woocommerce-ios/pull/10519#discussion_r1304335805), since there are a few PRs using this as a base. After resolving it, it should be good to go!

File: WooCommerce/WooCommerceTests/Mocks/MockCollectOrderPaymentAnalyticsTracker.swift (hunk 4 above)
Comment (joshheald) [reply WooCommerce/WooCommerceTests/Mocks/MockCollectOrderPaymentAnalyticsTracker.swift]: Ugh I don't know why I always get this wrong! Thank you.

File: WooCommerce/WooCommerceTests/Mocks/MockCardPresentPaymentPreflightController.swift (hunk 3 above)
Comment (joshheald) [reply WooCommerce/WooCommerceTests/Mocks/MockCardPresentPaymentPreflightController.swift:8]: I know what you mean, but the mock needs to match the protocol, and we don't actually need it to do anything. If we do in future, we can add it, but there's no harm in an empty async method any more than an empty sync method.

File: WooCommerce.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 1 above)
Comment (joshheald) [reply WooCommerce.xcworkspace/xcshareddata/swiftpm/Package.resolved]: Oh, good spot. Thank you. I've fixed it in ac65e90

**Review by iamgabrielma — APPROVED** [review]

**Pull request merged**